### PR TITLE
cleanup: add override when implementing an interface

### PR DESCRIPTION
**Pull request by Liuchang0812**
Fixes: http://tracker.ceph.com/issues/18922
Signed-off-by: liuchang0812 <liuchang0812@gmail.com>

### CODE REVIEW

**Comment by Liuchang0812** [comment]
jenkins test it please

**Comment by Liuchang0812** [comment]
jenkins test this please

**Comment by Liuchang0812** [comment]
```
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/ceph_objectstore_tool.py", line 1988, in <module>
    status = main(sys.argv[1:])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/ceph_objectstore_tool.py", line 1072, in main
    (pgid, coll, jsondict) = json.loads(JSONOBJ[0])[0]
IndexError: list index out of range

```

**Comment by liewegas** [comment]
retest this please

File: src/cls/rgw/cls_rgw_client.cc [hunk 1]
```diff
@@ -24,8 +24,8 @@ class ClsBucketIndexOpCtx : public ObjectOperationCompletion {
   int *ret_code;
 public:
   ClsBucketIndexOpCtx(T* _data, int *_ret_code) : data(_data), ret_code(_ret_code) { assert(data); }
-  ~ClsBucketIndexOpCtx() {}
-  void handle_completion(int r, bufferlist& outbl) {
+  ~ClsBucketIndexOpCtx() override {}
```
Comment (varadakari) [inline src/cls/rgw/cls_rgw_client.cc]: Does it make any difference here?

**Comment by varadakari** [comment]
would be easy to review, if you can split the PR into multiple per sub module like auth, client etc...

**Comment by Liuchang0812** [comment]
@varadakari  thanks for reviewing, I will do it.

File: src/cls/rgw/cls_rgw_client.cc (hunk 1 above)
Comment (Liuchang0812) [inline src/cls/rgw/cls_rgw_client.cc]: It's inserted automatically. I think it's useful because of that its parent class decl a virtual deconstructor.

Comment (cbodley) [inline src/cls/rgw/cls_rgw_client.cc]: can we just remove the destructor in this case?

Comment (Liuchang0812) [inline src/cls/rgw/cls_rgw_client.cc]: @cbodley I will have a try

Comment (Liuchang0812) [inline src/cls/rgw/cls_rgw_client.cc]: @cbodley @varadakari 

I will remove override from destructor.  FYI: https://github.com/isocpp/CppCoreGuidelines/pull/737

**Comment by Liuchang0812** [comment]
@liewegas @cbodley @varadakari 

Hi, I close this PR, and I will do some works as following:

1. remove swap files
2. split this PR into multiples like

 ```
liuchang@liuchang-ThinkPad-T450:~/ceph$ git branch 
  cleanup-cls
  cleanup-librados
  cleanup-librbd
  cleanup-mds
  cleanup-mgr
* cleanup-misc
  cleanup-mon
  cleanup-msg
  cleanup-os
  cleanup-osd
  cleanup-osdc
  cleanup-rgw
  cleanup-test
  cleanup-tools
 ```
3. remove override from destructor [TODO]

**Comment by Liuchang0812** [comment]
@liewegas @varadakari @cbodley @tchaikov 

I have fixed all issues. I created some PR based on subsystem. They are:

mgr: https://github.com/ceph/ceph/pull/13436
rbd: https://github.com/ceph/ceph/pull/13437
mds: https://github.com/ceph/ceph/pull/13438
mon: https://github.com/ceph/ceph/pull/13440
osd: https://github.com/ceph/ceph/pull/13439
rgw: https://github.com/ceph/ceph/pull/13441
librados: https://github.com/ceph/ceph/pull/13442
common: https://github.com/ceph/ceph/pull/13443

please tell me if anything need to do. very thanks

**Comment by joscollin** [comment]
Liuchang0812,

Is there anything pending in #13437 ? I see your comment - "OK. I will make a new PR soon. thanks".

**Comment by Liuchang0812** [comment]
@joscollin yes, there are #13691 #13692 etc, Commits are merged expect #13962

**Pull request closed without merging**